.github/workflows/pr.yml

```diff
@@ -0,0 +1,160 @@
+            --api-key ${{ secrets.ATLAS_PUBLIC_API_KEY }} \
+            --private-api-key ${{ secrets.ATLAS_PRIVATE_API_KEY }} \
+            --project-id ${{ secrets.ATLAS_PROJECT_ID }}
+  build-linux:
+    runs-on: ubuntu-latest
+    name: Build Linux
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+      - name: Setup Ninja
+        uses: seanmiddleditch/gha-setup-ninja@master
+      - name: Build Realm Dart for Linux
+        run: ./scripts/build-linux.sh
+      - name: Store artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: librealm-linux
+          path: binary/**
+          retention-days: 1
+
+  tests-linux:
+    runs-on: ubuntu-latest
+    name: Tests Linux
+    env:
+      BAAS_URL: https://realm-dev.mongodb.com
+      BAAS_CLUSTER: ${{ needs.baas-linux.outputs.clusterName }}
+      BAAS_API_KEY: ${{ secrets.ATLAS_PUBLIC_API_KEY }}
+      BAAS_PRIVATE_API_KEY: ${{ secrets.ATLAS_PRIVATE_API_KEY }}
+      BAAS_PROJECT_ID: ${{ secrets.ATLAS_PROJECT_ID}}
+    needs:
+      - baas-linux
+      - build-linux
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: false
+      - name: Fetch artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: librealm-linux
+          path: binary
+      - name : Setup Dart SDK
+        uses: dart-lang/setup-dart@main
+        with:
+          sdk: stable
+      - name: Install dependencies
+        run: dart pub get
+      - name: Run tests
+        run: |
+          dart test -r expanded -j 1 --test-randomize-ordering-seed random
+
+      # TODO: these should go away once we have a proper release workflow
+      - name: Archive binary
+        if: ${{ success() && github.event_name == 'push' && github.ref_type == 'tag' }}
+        run: |
+          dart run realm_dart archive --source-dir $(pwd)/binary/linux --output-file $ARCHIVE_PATH
+      - name: Release artifacts
+        if: ${{ success() && github.event_name == 'push' && github.ref_type == 'tag' }}
+        uses: ncipollo/release-action@v1
+        with:
+          allowUpdates:  true
+          artifactErrorsFailBuild: true
+          draft: true
+          omitNameDuringUpdate: true
+          prerelease: false
+          omitPrereleaseDuringUpdate: true
+          artifacts: ${{ env.ARCHIVE_PATH }}
+          body: "ADD RELEASE NOTES"
+          omitBodyDuringUpdate: true
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+  flutter-linux:
+    runs-on: ubuntu-latest
+    name: Flutter Tests Linux
+    needs:
+      - build-linux
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: false
+      - name: Setup GTK
+        run: |
+          sudo apt-get update -y
+          sudo apt-get install -y libgtk-3-dev xvfb
+      - name: Setup Ninja
+        uses: seanmiddleditch/gha-setup-ninja@master
```

Comment:
Why do we need Ninja here if we're using the artifact from the previous job already?

Reply:
I believe this is needed to compile the native flutter application: https://github.com/realm/realm-dart/blob/2c43087906deef2c1d25aa6546afc329b70271ae/.github/workflows/pr.yml#L143

Reply:
_Interesting_. I guess Flutter doesn't ship precompiled binaries yet like Electron does.